python/pyiceberg/table/__init__.py

```diff
@@ -314,11 +314,19 @@ def _check_content(file: DataFile) -> DataFile:
         return file
 
 
-def _open_manifest(io: FileIO, manifest: ManifestFile, partition_filter: Callable[[DataFile], bool]) -> List[FileScanTask]:
+def _open_manifest(
+    io: FileIO,
+    schema: Schema,
+    row_filter: BooleanExpression,
+    manifest: ManifestFile,
+    partition_filter: Callable[[DataFile], bool],
+    case_sensitive: bool,
+) -> List[FileScanTask]:
     all_files = files(io.new_input(manifest.manifest_path))
     matching_partition_files = filter(partition_filter, all_files)
     matching_partition_data_files = map(_check_content, matching_partition_files)
-    return [FileScanTask(file) for file in matching_partition_data_files]
+    evaluator = _InclusiveMetricsEvaluator(schema, row_filter, case_sensitive)
```

Comment:
Do we need to create a new metrics evaluator for each manifest? The table schema, row filter, and case sensitive values are always the same.

I think we could handle this like `partition_filter` and pass in a `Callable`.

Reply:
Oeh, I like that _very_ much! Thanks for the sharp observation!